kcidb/monitor/spool/__init__.py

```diff
@@ -157,14 +175,17 @@ def pick(self, id, timestamp=None, timeout=None):
         def pick_if_not_picked(transaction, doc, timestamp):
             """Pick notification, if not picked yet"""
             # Get the document snapshot
-            snapshot = doc.get(field_paths=["picked_until", "message"],
+            snapshot = doc.get(field_paths=["picked_until", "message",
+                                            "due"],
```

Comment:
Could you not wrap this line, as well? It fits into 78 characters, which is fine.